js/index.js

```diff
@@ -13,3 +13,38 @@ for (let i=0; i<skills.length; i++) {
   skill.innerHTML=skills[i];
   skillsList.appendChild(skill);
 };
+const messageForm = document.getElementsByName('leave_message');
+messageForm[0].addEventListener('submit', (event) => {
+  event.preventDefault();
+  const name = event.target.name;
+  const email = event.target.email;
+  const message = event.target.message;
+  console.log(name.value);
+  console.log(email.value);
+  console.log(message.value);
+
+  messageSection = document.getElementById('messages');
+  messageList = messageSection.querySelector('ul');
+  newMessage = document.createElement('li');
+
+  const a=document.createElement('a');
+  const linkText = document.createTextNode(`${name.value}`);
+  a.appendChild(linkText);
+  a.title = `${name.value}`;
+  a.href = `mailto:${email.value}`;
+  newMessage.appendChild(a);
+  newMessage.insertAdjacentHTML('beforeend', `<span> wrote: ${message.value} </span`);
+// creating a new element, a button
+  removeButton = document.createElement('button');
+  removeButton.innerHTML = 'remove';
+  removeButton.type = 'button';
+  messageList.appendChild(newMessage);
+  newMessage.appendChild(removeButton);
+
+  removeButton.addEventListener('click', (event) =>{
+    const entry = removeButton.parentNode;
```

Comment:
So this is _almost_ perfect. You might notice if you run your code as-is, enter 2-3 messages, and then try to delete one, things get a little wonky. It removes the very last one no matter which remove button you click, and all remaining remove buttons then stop working. How to fix it?  You want:

`const entry = event.target.parentNode`

CTD's instructions for that part are a little confusing!  You do need to find the button's parent element via DOM traversal as they say. But for it to associate properly with what you click, you need to start that traversal from the click event.

LMK (or the mentors) if you have any questions! I'm on the CTD Slack.